### PR TITLE
Basic Support for Scsb message queue errors and availability

### PR DESCRIPTION
**Pull request by kevinreiss**
@tampakis. Keeping this open as a WIP for now.

### CODE REVIEW

File: app/views/scsb_mailer/_message_details.text.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<% if @success == true %>
+  <%= "Title: #{@message[:titleIdentifier]}" %>
+  <%= "Barcode: #{@message[:itemBarcode]}" %>
+  <% unless @message[:requestNotes].blank? %>
+    <%= "Notes: #{@message[:requestNotes]}" %>
+  <% end %>
+
+  <%= "Your Library Account: #{link_to 'https://pulsearch.princeton.edu/account', 'https://pulsearch.princeton.edu/account'}" %>
```
Comment (tampakis) [inline app/views/scsb_mailer/_message_details.text.erb]: You can remove the second arg `https://pulsearch.princeton.edu/account` if the url and display text are the same

File: config/locales/scsb.en.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+en:
+  scsb:
+    default_from: 'fstcirc@princeton.edu'
+    default_error_to: 'kr2@princeton.edu'
+    request:
+      email_subject: "Offsite Request Confirmation"
+      email_fail_subject: 'Offsite Request Failure'
+      delivery_label: "Physical Item Delivery"
+      email_conf_msg: "Your will receive an email in 1-2 business days when your item is delivered and ready for pickup."
```
Comment (tampakis) [inline config/locales/scsb.en.yml]: You will receive

File: config/locales/scsb.en.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    edd:
+      email_subject: "Recap Electronic Delivery Confirmation"
+      email_fail_subject: 'Offsite EDD Request Failure'
+      brief_msg: "Item offsite at ReCAP Facility. Request a selection be scanned for you and delivered as a PDF file in 1-2 business days."
+      email_conf_msg: "Electronic copy requests typically take 1-2 days to process. You will receive an email including a link where you can download your scanned selection in the that timeframe."
```
Comment (tampakis) [inline config/locales/scsb.en.yml]: within that timefrime

File: config/locales/scsb.en.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    recall:
+      success: "You have successfully placed a recall. The item may not be available for pickup for several weeks, at which point you will be notified. See confirmation email with details about when your item(s) will be available."
```
Comment (tampakis) [inline config/locales/scsb.en.yml]: item vs item(s) should be consistent

File: config/locales/scsb.en.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      email_subject: "Recap Recall Confirmation"
+      email_fail_subject: "Recap Recall Request Failure"
+      email_conf_msg: "Your are requesting that another Library user return this book for your use. This process can take up to several weeks. Contact the Library Circulation Desk for Details. If possible please request the material in question via Borrow Direct or Interlibrary Loan."
```
Comment (tampakis) [inline config/locales/scsb.en.yml]: lowercase details

File: config/locales/scsb.en.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    default_error_to: 'kr2@princeton.edu'
+    request:
+      email_subject: "Offsite Request Confirmation"
+      email_fail_subject: 'Offsite Request Failure'
+      delivery_label: "Physical Item Delivery"
+      email_conf_msg: "Your will receive an email in 1-2 business days when your item is delivered and ready for pickup."
+    edd:
+      email_subject: "Recap Electronic Delivery Confirmation"
+      email_fail_subject: 'Offsite EDD Request Failure'
+      brief_msg: "Item offsite at ReCAP Facility. Request a selection be scanned for you and delivered as a PDF file in 1-2 business days."
```
Comment (tampakis) [inline config/locales/scsb.en.yml]: Lowercase facility

File: spec/jobs/scsb_request_job_spec.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    described_class.new.perform(fail_message)
+    expect(described_class.queue_name?).to be true
+    expect(described_class.queue_name).to eq('scsb_request')
+  end
+
+  # it 'logs the message passed to the job' do
+  #   described_class.new.perform(fail_message)
+  #   allow(described_class.logger).to receive(:info).and_call_original
+  #   expect(described_class.logger).to receive(:info).with("Processing Message #{fail_message}")
+  # end
```
Comment (tampakis) [inline spec/jobs/scsb_request_job_spec.rb]: Can you remove the commented out specs?

File: app/mailers/scsb_mailer.rb [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  def export_email(message)
+      @message = message
+      destination_email = 'lsupport@princeton.edu'
+      mail(to: destination_email,
+           subject: subject_line("Scsb Partner Export"))
+  end
+
+  def error_email(args)
```
Comment (tampakis) [inline app/mailers/scsb_mailer.rb]: does this method need to be here?

File: spec/jobs/scsb_request_job_spec.rb (hunk 7 above)
Comment (kevinreiss) [inline spec/jobs/scsb_request_job_spec.rb]: Get a weird action view error when I remove it
```      Failure/Error: <%= "Your Library Account: #{link_to 'https://pulsearch.princeton.edu/account'}" %>

      ActionView::Template::Error:
        No route matches {:action=>"index"}
```

File: app/views/scsb_mailer/_message_details.text.erb (hunk 1 above)
Comment (kevinreiss) [inline app/views/scsb_mailer/_message_details.text.erb]: Get a weird action view error when I remove it
```      Failure/Error: <%= "Your Library Account: #{link_to 'https://pulsearch.princeton.edu/account'}" %>

      ActionView::Template::Error:
        No route matches {:action=>"index"}
```

Comment (tampakis) [inline app/views/scsb_mailer/_message_details.text.erb]: oh sorry about that, i misread the documentation for the method

**Pull request merged**